### DISCUSSION TITLE
Is there an event to listen for a new CssRule?

### DISCUSSION THREAD

**Discussion by raywong-fb** [GrapesJS/grapesjs#4351] [Q&A]
Hi!

I'm using `CssComposer.setRule()` to add/update styles. Is there any way to listen for an event that will trigger when a new rule has been added?

For existing rules, I'm able to listen to each rules `change:style` event when the rule has been updated. But I'm not sure what event to listen to if a new rule has been added using `setRule`.

Thanks!

**Comment by senthil177cse** [comment]
Hi friend...
do you how to add external styles in grapesjs...
Fors Ex: if i am adding bootstrap.css file in canvas means styles sppending bootstrap classes but editor.getCss() code not getting the bootstrap classes ... If you know answer please help me...

**Comment by ronaldohoch** [comment]
What are you trying to do?

**Reply by raywong-fb** [reply]
Essentially, I'm trying to listen each time `setRule` is called. My workaround for now is to run a custom command each time `setRule` is called, and then listen for that custom command. I was just curious if there's a built-in event I could make use of instead of a custom command.

**Comment by Ju99ernaut** [comment]
I don't think that's available from the API, but you can probably get the rules model and subscribe to it's `...on('add', ...)` event since it probably extends a backbone  model.
